_posts/en/2019-10-21-schnorr-taproot-workshop.md

```diff
@@ -0,0 +1,189 @@
+{:.center}
+
+Elichai Turkel gives a refresher on the basic elliptic curve math that will be
+required for this workshop.
+
+#### 0.3 Tagged Hashes
+
+_(No video)_ This chapter introduces _Tagged hashes_, which are used in both
+the bip-schnorr and bip-taproot proposals.
+
+## Schnorr signatures and MuSig
+
+This section explains the bip-schnorr proposal and explains how MuSig can be
+used to aggregate multiple public keys and partial signatures into a single
+pubkey and signature.
+
+#### 1.1 Schnorr Signatures
+
+[![Schnorr Signatures](/img/posts/taproot-workshop/schnorr.png)](https://www.youtube.com/watch?v=wybiVFdknhg&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=3)
+{:.center}
+
+Elichai explains the mathematics behind schnorr signatures and explains the
+bip-schnorr proposal.
+
+#### 1.2 MuSig
+
+[![MuSig](/img/posts/taproot-workshop/musig.png)](https://www.youtube.com/watch?v=5MbTptrXEC4&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=4)
+{:.center}
+
+Elichai describes the [MuSig algorithm][musig] (authored by Gregory Maxwell,
+Andrew Poelstra, Yannick Seurin and Pieter Wuille), and shows how it can be
+used to aggregated multiple public keys and partial signatures into a single
+pubkey/signature.
+
+## Taproot
+
+This section explains the bip-taproot and bip-tapscript proposals. It shows the
+format of a segwit v1 output and how such an output can be spent in a key path
+spend or a script path spend. It shows how a tweaked public key can commit to
+one or more scripts, and how the segwit v1 output can be spent using one of those
+scripts.
+
+#### 2.0 Introduction to Taproot
+
+[![Introduction to Taproot](/img/posts/taproot-workshop/taproot-intro.png)](https://www.youtube.com/watch?v=KLNH0ttpdFg&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=5)
+{:.center}
+
+James Chiang gives an overview of the bip-taproot/bip-tapscript proposals. This
+notebook demonstrates how we'll create transaction outputs, then spend them and
+verify that the spend is valid.
+
+#### 2.1 Segwit Version 1
+
+[![Segwit Version 1](/img/posts/taproot-workshop/segwit-version-1.png)](https://www.youtube.com/watch?v=n-jAUaSkcAA&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=6)
+{:.center}
+
+James shows how to create segwit version 1 transaction outputs and spend them
+using the key path spend.
+
+#### 2.2 Taptweak
+
+[![Taptweak](/img/posts/taproot-workshop/taptweak.png)](https://www.youtube.com/watch?v=EkGbPxAExdQ&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=7)
+{:.center}
+
+James explains what a key tweak is, and how a tweak can be
+used to commit to arbitrary data.
+
+#### 2.3 Tapscript
+
+[![Tapscript](/img/posts/taproot-workshop/tapscript.png)](https://www.youtube.com/watch?v=nXGe9_M5pjk&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=8)
+{:.center}
+
+James shows how using a taptweak, we can commit a tapscript in a segwit v1
```

Comment:
idem re "shows"

"taptweak" could be read as tapt weak or tap tweak, perhaps separate the two words unless this is an established word?

Reply:
yes, I find "TapTweak" in the BIP to be clearer with the capital letters separation

Reply:
Our style guide states that `taproot` is a common noun and uncapitalized. For consistency, we therefore use `tapscript`, `taptree`, `taptweak`, etc.